libvmi/driver/xen/xen_events_48.c

```diff
@@ -562,12 +574,30 @@ status_t process_privcall_event(vmi_instance_t vmi,
     event->arm_regs = (arm_registers_t *)&req->data.regs.arm;
     event->slat_id = (req->flags & VM_EVENT_FLAG_ALTERNATE_P2M) ? req->altp2m_idx : 0;
     event->vcpu_id = req->vcpu_id;
+    event->interrupt_event.offset = req->data.regs.arm.pc & VMI_BIT_MASK(0,11);
+    event->interrupt_event.gla = req->data.regs.arm.pc;
+
+    if ( VMI_FAILURE == vmi_translate_kv2p(vmi, req->data.regs.arm.pc, &event->interrupt_event.gfn) ) {
+        errprint("%s: cannot translate pc to physical address\n", __FUNCTION__);
+        return VMI_FAILURE;
+    }
+    event->interrupt_event.gfn = event->interrupt_event.gfn >> 12;
```

Comment:
Instead of stuffing these infos into the `interrupt_event` structure, please define a new structure in `libvmi/events.h` to hold them.

Reply:
I absolutely agree. Ulrich, as already mentioned before, privcalls are not interrupts. As such, both structures must be differentiated in the code, as well.

Reply:
Thanks and remember you can just push to this branch ;)